specs/governance/gov-token.md

```diff
@@ -24,72 +24,87 @@
+in the `_afterTokenTransfer` function. If the call to the Alligator's `afterTokenTransfer` function fails, the token transfer
+will still be completed, and an event will be emitted to indicate the failure. This prevents the token transfer from being
+blocked due to issues with the Alligator contract.
 
 ```solidity
-struct Checkpoint {
-    uint32 fromBlock;
-    uint224 votes;
+function _afterTokenTransfer(address from, address to, uint256 amount) internal override {
+    super._afterTokenTransfer(from, to, amount);
+    try IAlligator(alligatorAddress).afterTokenTransfer(from, to, amount) {
+        // Call successful, no action needed
+    } catch {
+        // Call failed, emit an event to indicate the failure
+        emit AlligatorCallFailed(from, to, amount);
+    }
 }
 ```
 
-Checkpoints are organized as a one-to-one mapping of user addresses to append-only lists of checkpoints. When a
-user is involved in a token transfer, mint, burn, or vote delegation, a new checkpoint is added to their corresponding
-list of checkpoints, reflecting the user's updated voting power at that block number.
+All delegation-related state, including the `delegates`, `checkpoints`, and `numCheckpoints` mappings, is shifted from the
+`GovernanceToken` to the `Alligator` contract. The `Alligator` treats the original checkpoints from the `GovernanceToken`
+as a starting point for its own checkpoints.
 
-```solidity
-mapping(address => Checkpoint[]) private _checkpoints;
-```
-
-For any checkpoint list, the `fromBlock` value of each checkpoint is strictly increasing as a factor of the checkpoint's
-index in the list. Additionally, if a checkpoint is created when there's already a checkpoint at the same block number,
-the new checkpoint replaces the old one.
+Basic delegation functions in the `GovernanceToken`, such as `delegate` and `delegateBySig`, are modified to forward the
+calls to the `Alligator` contract using an advanced delegation rule that mimics basic delegation.
 
 ### Token Minting
 
 `GovernanceToken` MUST have a `mint(address,uint256)` function with external visibility that allows the contract owner
-to mint an arbitrary number of new tokens to an specific address. This function MUST only be called by the contract
+to mint an arbitrary number of new tokens to a specific address. This function MUST only be called by the contract
 owner, the `MintManager`, as enforced by the `onlyOwner` modifier inherited from the `Ownable` contract. When tokens
-are minted, the voting power of the recipient address MUST be updated accordingly. The total token supply is capped to
-`2^208^ - 1` to prevent overflow risks in the voting system. If the total supply exceeds this limit,
-`_mint(address,uint256)`, as inherited from `ERC20Votes`, MUST revert.
+are minted, the voting power of the recipient address MUST be updated accordingly in the `Alligator` contract. The total
```

Comment:
should we specify that the mints and burns are updated in the Alligator via the transfer hooks? (minting transfers from `address(0)`, and burns transfers to `address(0)`)